### PR TITLE
fix: Pass uiSchema to custom ArrayField

### PR DESCRIPTION
**Pull request by bakajvo**
### Reasons for making this change

This is similar issue like [PR 2587](https://github.com/rjsf-team/react-jsonschema-form/pull/2587) in BooleanFields. This PR fixes passing UISchema to ArrayFields.

### Checklist
- [x]  **I'm adding or updating code**
  - [x] I've added and/or updated tests

### CODE REVIEW

**Comment by bakajvo** [comment]
Follow up on [Issue 2793](https://github.com/rjsf-team/react-jsonschema-form/issues/2793)

File: packages/core/test/ArrayField_test.js [hunk 1]
```diff
@@ -1838,6 +1838,41 @@ describe("ArrayField", () => {
...
+          custom_field_key: {
+            "ui:options": {
+              test: "foo",
+            },
+          },
+        },
+        formData: ["foo", "bar"],
+      });
+
+      expect(node.querySelector("#custom-ui-option-value").textContent).to.eql(
```
Comment (epicfaace) [inline packages/core/test/ArrayField_test.js:1871]: The current test only covers renderCustomWidget.

can you also add tests that cover
- renderNormalArray
- renderFixedArray
- renderFiles
- renderMultiSelect?

Comment (bakajvo) [reply packages/core/test/ArrayField_test.js:1963]: Done, can u please check it?

**Comment by bakajvo** [comment]
Hello, can @epicfaace review it please? :)

**Review by epicfaace — APPROVED** [review]
Nice! These tests are 💯

**Comment by epicfaace** [comment]
@bakajvo I think you need to run the format / prettier command though.

**Comment by bakajvo** [comment]
> @bakajvo I think you need to run the format / prettier command though.

@epicfaace Done, thx! :)

**Pull request merged**